boa/src/environment/mod.rs

```diff
@@ -6,3 +6,31 @@ pub mod function_environment_record;
 pub mod global_environment_record;
 pub mod lexical_environment;
 pub mod object_environment_record;
+
+#[derive(Debug)]
+pub enum ErrorKind {
+    ReferenceError(String),
+    TypeError(String),
+}
+
+use crate::builtins::value::Value;
+use crate::exec::Interpreter;
+
+impl ErrorKind {
+    pub fn to_error(&self, interpreter: &mut Interpreter) -> Value {
+        match self {
+            ErrorKind::ReferenceError(msg) => interpreter.construct_reference_error(msg),
+            ErrorKind::TypeError(msg) => interpreter.construct_type_error(msg),
+        }
+    }
+}
+
+#[macro_export]
+macro_rules! try_error {
+    ($e: expr) => {
+        match $e {
+            Err(e) => return Err(e.to_error(interpreter)),
+            _ => (),
+        }
+    };
+}
```

Comment:
why is it exported we don't use it out side of the crate

Reply:
I shouldn't have added this. This was just a temporary thing.